test/e2e/localstorage_test.go

```diff
@@ -139,6 +139,135 @@ func TestLocalStorageOperator(t *testing.T) {
+	selectedNode.Spec.Taints = []v1.Taint {
+		{
+			Key:	"localstorage",
+			Value:	"testvalue",
+			Effect: "NoSchedule",
+		},
+	}
+	updatedNode, err := waitForNodeTaintUpdate(t, f.KubeClient, selectedNode, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("error tainting node : %v", err)
+	}
+	selectedNode = updatedNode
+
+	selectedDisk, _ := selectDisk(f.KubeClient, selectedNode)
+	waitForPVCreation := true
+	if selectedDisk == "" {
+		waitForPVCreation = false
+		selectedDisk = "/dev/foobar"
+	}
+	localVolume := &localv1.LocalVolume{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "LocalVolume",
+			APIVersion: localv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-local-disk",
+			Namespace: namespace,
+		},
+		Spec: localv1.LocalVolumeSpec{
+			Tolerations: []v1.Toleration{
+				{
+					Key:	"localstorage",
+					Value:	"testvalue",
+					Operator: "Equal",
+				},
+			},
+			NodeSelector: &v1.NodeSelector{
+				NodeSelectorTerms: []v1.NodeSelectorTerm{
+					{
+						MatchFields: []v1.NodeSelectorRequirement{
+							{Key: "metadata.name", Operator: v1.NodeSelectorOpIn, Values: []string{selectedNode.Name}},
+						},
+					},
+				},
+			},
+			StorageClassDevices: []localv1.StorageClassDevice{
+				{
+					StorageClassName: "test-local-sc",
+					DevicePaths:      []string{selectedDisk},
+				},
+			},
+		},
+	}
+
+	err = f.Client.Create(goctx.TODO(), localVolume, &framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		t.Fatalf("error creating localvolume cr : %v", err)
+	}
+
+	provisionerDSName := localVolume.Name + "-local-provisioner"
+	diskMakerDSName := localVolume.Name + "-local-diskmaker"
+
+	err = waitForDaemonSet(t, f.KubeClient, namespace, provisionerDSName, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("error waiting for provisioner daemonset : %v", err)
+	}
+
+	err = waitForDaemonSet(t, f.KubeClient, namespace, diskMakerDSName, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("error waiting for diskmaker daemonset : %v", err)
+	}
+
+	err = verifyLocalVolume(localVolume, f.Client)
+	if err != nil {
+		t.Fatalf("error verifying localvolume cr: %v", err)
+	}
+
+	err = checkLocalVolumeStatus(localVolume)
+	if err != nil {
+		t.Fatalf("error checking localvolume condition: %v", err)
+	}
+
+	if waitForPVCreation {
+		err = waitForCreatedPV(f.KubeClient, localVolume)
+		if err != nil {
+			t.Fatalf("error waiting for creation of pv: %v", err)
+		}
+		err = deleteCreatedPV(f.KubeClient, localVolume)
+		if err != nil {
+			t.Errorf("error deleting created PV: %v", err)
+		}
+	}
+
+	// Restore the original set of node taints after the test is completed.
+	selectedNode.Spec.Taints = nodeTaints
+	selectedNode, err = waitForNodeTaintUpdate(t, f.KubeClient, selectedNode, retryInterval, timeout)
+	if err != nil {
+		t.Fatalf("error restoring original taints on node : %v", err)
+	}
+}
```

Comment:
This test as a separate function IMO adds very little to existing test suite and copies large chunk of code from previous function. I would think that we can reuse existing function to add checks for tolerations too.  It would help keeping test time shorter and test file shorter too.

The other thing is - the test should actually verify if created daemonset has user specified toleration which it does not currently.

Reply:
The test has been updated to run through all of the elements defined at the start of the function, and it confirm that the DaemonSets' Tolerations match the LocalVolume's Tolerations.